internal/jobcontainers/path_test.go

```diff
@@ -31,52 +33,159 @@ func TestSearchPath(t *testing.T) {
 	}
 }
 
-func TestGetApplicationName(t *testing.T) {
+func TestGetApplicationNamePing(t *testing.T) {
 	expected := "C:\\WINDOWS\\system32\\ping.exe"
 
 	cwd, err := os.Getwd()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	path, _, err := getApplicationName("ping", cwd, os.Getenv("PATH"))
+	pathEnv := os.Getenv("PATH")
+
+	path, _, err := getApplicationName("ping", cwd, pathEnv)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assertStr(t, expected, path)
+
+	path, _, err = getApplicationName("./ping", cwd, pathEnv)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertStr(t, expected, path)
 
-	path, _, err = getApplicationName("./ping", cwd, os.Getenv("PATH"))
+	path, _, err = getApplicationName(".\\ping", cwd, pathEnv)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertStr(t, expected, path)
 
-	path, _, err = getApplicationName(".\\ping", cwd, os.Getenv("PATH"))
+	// Test relative path with different cwd
+	newCwd := `C:\Windows\`
+	_, _, err = getApplicationName("./system32/ping", newCwd, pathEnv)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertStr(t, expected, path)
 
+	pingWithCmd := "cmd /c ping 127.0.0.1"
+	path, cmdLine, err := getApplicationName("cmd /c ping 127.0.0.1", cwd, pathEnv)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assertStr(t, cmdLine, pingWithCmd)
+	assertStr(t, "C:\\windows\\system32\\cmd.exe", path)
+
 	// Test that we only find the first element of the commandline if the binary exists.
-	path, _, err = getApplicationName("ping test", cwd, os.Getenv("PATH"))
+	path, _, err = getApplicationName("ping test", cwd, pathEnv)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertStr(t, expected, path)
 
 	// Test quoted application name with an argument afterwards.
-	path, cmdLine, err := getApplicationName("\"ping\" 127.0.0.1", cwd, os.Getenv("PATH"))
+	path, cmdLine, err = getApplicationName("\"ping\" 127.0.0.1", cwd, pathEnv)
 	if err != nil {
 		t.Fatal(err)
 	}
 	assertStr(t, expected, path)
 
-	args := splitArgs(cmdLine)
 	cmd := &exec.Cmd{
 		Path: path,
-		Args: args,
+		Args: splitArgs(cmdLine),
 	}
 	if err := cmd.Run(); err != nil {
 		t.Fatal(err)
 	}
 }
+
+func TestGetApplicationNameRandomBinary(t *testing.T) {
+	pathEnv := os.Getenv("PATH")
```

Comment:
We shouldn't rely on the environment's `PATH` for unit test code, as that could easily cause it to break in different environments.

Reply:
I'm blindly assuming PATH will at least have system32 in it for the ping tests above. For these random binary tests, PATH is mostly irrelevant and I can just use an empty string.